File: 프로그래머스/2/77885. 2개 이하로 다른 비트/2개 이하로 다른 비트.java
class Solution {
    public long[] solution(long[] numbers) {
        long[] answer = new long[numbers.length];
        for (int i = 0; i < numbers.length; i++) {
            answer[i] = f(numbers[i]);
        }
        return answer;
    }
    
    public long f(long x) {
        long result = x;
        long n = 0;
        while (x % 2  == 1) {
            n += 1;
            x /= 2;
        }
        if (n != 0) { n -= 1; }
        return result + (1L<<n);
    };
}

File: 프로그래머스/2/77885. 2개 이하로 다른 비트/2개 이하로 다른 비트.py
def f(x):
    n = 0
    num = x
    while num % 2:
        n += 1
        num //= 2
    if (n != 0) :
        n -= 1
    
    return x + (2**n)
    
def solution(numbers):
    answer = []
    for num in numbers:
        answer.append(f(num))
    return answer